cdap-ui/app/services/constants.js

```diff
@@ -48,8 +58,38 @@ angular.module(PKG.name + '.services')
         'httpcallback': 'Make HTTP Call'
       }
     },
+    'pluginTypeToID': {
+      'transform': 'Transform',
+      'batchsource': 'BatchSource',
+      'batchsink': 'BatchSink',
+      'batchaggregator': 'Aggregate',
+      'realtimesink': 'RealtimeSink',
+      'realtimesource': 'RealtimeSource',
+      'sparksink': 'Model',
+      'sparkcompute': 'Compute',
+      'batchjoiner': 'Join',
+      'action': 'Action',
+      'streamingsource': 'BatchSource',
+      'windower': 'Windower'
```

Comment:
Its just a separation of label from id. For instance 'Sink' is represented as both 'BatchSink' and 'RealtimeSink'. This ID map is to differentiate between the two. ID could be 'BatchSink' or 'RealtimeSink' & the label could be just 'Sink'.